feat: Prioritize Search icon, move Shadowbox to overflow

PrimaryVision opened the pull request:
This PR introduces a new setting in General Settings (Prioritise search icon in the toolbar) to customize the main toolbar layout, addressing feature request [#97](https://github.com/edgan/Slide/issues/97). When enabled, this setting modifies the toolbar to prioritize the Search, Sort, and Refresh icons by moving the Shadowbox icon to the overflow menu. 

Note: The actual number of prioritized icons visible on the toolbar depends on the device's display size setting; default settings may show fewer icons before they move to the overflow menu.

edgan commented:
It seems very close to the `Expanded toolbar menu` option. Which for me adds the `Refresh` button. Your `Prioritise search` icon in the toolbar doesn't just move the `Shadowbox mode` to the menu, and the `Search box` to the front of the list. It also seems to act like `Expanded toolbar menu` and shows the `Refresh` button.

Normal:
![image](https://github.com/user-attachments/assets/46c4e803-acae-4e17-baba-9e57ed2b5dea)

Expanded:
![image](https://github.com/user-attachments/assets/d78c9c11-443f-4c65-bcb0-a47a3d05b9aa)

Prioritise search:
![image](https://github.com/user-attachments/assets/50dded25-6b12-4b31-b96c-ceca473062c0)

PrimaryVision commented:
The `expanded toolbar` appears weird for users comfortable with the standard density settings, which likely includes most users. It might look less odd with a denser UI, but I doubt users would adjust their global UI settings just for one app.

PrimaryVision commented on app/src/main/res/values/strings.xml:
```diff
@@ -719,6 +719,8 @@
     <string name="settings_layout_picture_mode">Picture mode</string>
     <string name="settings_general_expanded_toolbar">Expanded toolbar menu</string>
     <string name="settings_general_expanded_toolbar_desc">Show more options on the subreddit toolbar</string>
+    <string name="settings_general_toolbar_layout_title">Prioritise search icon in the toolbar</string>
```
Fixed the spelling, changed to `Prioritize`.

PrimaryVision commented:
Simplified the branch to align with the original intent of prioritizing the `Search` icon. Removed the previous settings logic for selectable toolbar layouts.
This PR now only updates the default toolbar menu files:
- Prioritize the Search, Sort, and Refresh icons (shown if room).
- Move the other icons to the overflow menu.

edgan commented:
Yeah, this is so much simpler. :+1: 

These two need to be updated for consistency:
```
app/src/main/res/menu/menu_single_subreddit_expanded.xml:
1. shadowbox
2. sort
3. search
4. refresh

app/src/main/res/menu/menu_subreddit_overview_expanded.xml:
1. shadowbox
2. sort
3. search
4. refresh
```

Already in the PR, and given as an example:
```
app/src/main/res/menu/menu_single_subreddit.xml:
1. search
2. sort
3. refresh

app/src/main/res/menu/menu_subreddit_overview.xml:
1. search
2. sort
3. refresh
```

PrimaryVision commented:
Updated the expanded toolbar menu files to set the icon order: Search, Sort, Refresh, Shadowbox, and Gallery (if room), with remaining icons in the overflow menu.
